metric/metric.go

```diff
@@ -8,20 +8,66 @@ import (
 	"github.com/prometheus/client_golang/prometheus/promauto"
 )
 
+const namespace = "gatus" // The prefix of the metrics
+
 var (
-	// This will be initialized once PublishMetricsForEndpoint.
-	// The reason why we're doing this is that if metrics are disabled, we don't want to initialize it unnecessarily.
-	resultCount *prometheus.CounterVec = nil
+	initializedMetrics bool // Whether the metrics have been initialized
+
+	resultTotal                        *prometheus.CounterVec
+	resultDurationSeconds              *prometheus.GaugeVec
+	resultConnectedTotal               *prometheus.CounterVec
+	resultCodeTotal                    *prometheus.CounterVec
+	resultCertificateExpirationSeconds *prometheus.GaugeVec
 )
 
+func initializePrometheusMetrics() {
+	resultTotal = promauto.NewCounterVec(prometheus.CounterOpts{
+		Namespace: namespace,
+		Name:      "results_total",
+		Help:      "Number of results per endpoint",
+	}, []string{"key", "group", "name", "type", "success"})
+	resultDurationSeconds = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: namespace,
+		Name:      "results_duration_seconds",
+		Help:      "Duration of the request in seconds",
+	}, []string{"key", "group", "name", "type"})
+	resultConnectedTotal = promauto.NewCounterVec(prometheus.CounterOpts{
+		Namespace: namespace,
+		Name:      "results_connected_total",
+		Help:      "Total number of results in which a connection was successfully established",
+	}, []string{"key", "group", "name", "type"})
+	resultCodeTotal = promauto.NewCounterVec(prometheus.CounterOpts{
+		Namespace: namespace,
+		Name:      "results_code_total",
+		Help:      "Total number of results by code",
+	}, []string{"key", "group", "name", "type", "code"})
```

Comment:
FYI I merged `results_dns_return_code_total` and `results_http_status_code_total` into `results_code_total` 3193b2c because they're effectively the same, and you can separate them by looking at their `type`.

Reply:
this change is good for reduce metrics nums.
good job :)